wandb/integration/openai/resolver.py

```diff
@@ -145,5 +180,84 @@ def _request_response_result_to_trace(
             )
             for choice in choices
         ]
-        trace = self.results_to_trace_tree(request, response, results, time_elapsed)
-        return trace
+        metrics = self._get_metrics_to_log(request, response, results, time_elapsed)
+        return metrics
+
+    def _get_usage_metrics(
+        self,
+        response: Response,
+        time_elapsed: float,
+        model_alias_and_cost: Optional[Dict[str, Any]] = None,
+    ) -> Dict[str, Any]:
+        """Gets the usage stats from the response object."""
+        usage_stats = {}
```

Comment:
I would turn this into a dataclass so that it's easy to customize in the future.